From ac5865730e46901f1d56e6ae4fd916c2327d36ce Mon Sep 17 00:00:00 2001
From: Prarit Bhargava <prarit@redhat.com>
Date: Thu, 28 Aug 2014 14:21:59 -0400
Subject: [cpufreq] remove unnecessary check in __cpufreq_governor()

Message-id: <1409235879-1283-26-git-send-email-prarit@redhat.com>
Patchwork-id: 89365
O-Subject: [RHEL7.1 PATCH BZ 1134639 025/185] cpufreq: remove unnecessary check in __cpufreq_governor()
Bugzilla: 1134369
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: Mark Langsdorf <mlangsdo@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1134639

commit 9e9fd801676a946b759a8669baa24ba327c8c903
Author: Viresh Kumar <viresh.kumar@linaro.org>
Date:   Tue Aug 20 12:08:24 2013 +0530

    cpufreq: remove unnecessary check in __cpufreq_governor()

    We don't need to check if event is CPUFREQ_GOV_POLICY_INIT and put
    governor module as we are sure event can only be START/STOP here.

    Remove the useless check.

    Signed-off-by: Viresh Kumar <viresh.kumar@linaro.org>
    Signed-off-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>

Cc: Lenny Szubowicz <lszubowi@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/cpufreq/cpufreq.c b/drivers/cpufreq/cpufreq.c
index 933be44..ff1d9a7 100644
--- a/drivers/cpufreq/cpufreq.c
+++ b/drivers/cpufreq/cpufreq.c
@@ -1702,8 +1702,6 @@ static int __cpufreq_governor(struct cpufreq_policy *policy,
  if ((!policy->governor_enabled && (event == CPUFREQ_GOV_STOP)) ||
      (policy->governor_enabled && (event == CPUFREQ_GOV_START))) {
   mutex_unlock(&cpufreq_governor_lock);
-  if (event == CPUFREQ_GOV_POLICY_INIT)
-   module_put(policy->governor->owner);
   return -EBUSY;
  }
 
-- 
1.7.1